src/context/TimerContext.tsx

```diff
@@ -0,0 +1,296 @@
+import { createContext, useState, useEffect, useRef, useContext } from 'react'
+import { Timer } from '../types/timers'
+import { formatTime } from '../utils/helpers'
+
+// Global context for Timer
+export const TimerContext = createContext({
+    timers: [] as Timer[],
+    timeInMs: 0,
+    running: false,
+    currentTimer: null as Timer | null,
+    currentTimerIndex: 0,
+    currentPhase: 'Work',
+    currentRound: 1,
+    addTimer: (_timer: Timer) => {},
+    removeTimer: (_index: number) => {},
+    updateTimer: (_index: number, _timer: Timer) => {},
+    toggleRunning: () => {},
+    fastForward: () => {},
+    resetWorkout: () => {}
+})
+
+// Get timer details and display it within queue
+export const displayTimerDetails = (timer: Timer) => {
+    const { type, settings } = timer
+    
+    let details = `${type}: `;
+
+    if (settings.rounds) {
+      details += `${settings.rounds} Rounds x `
+    }
+    if (settings.totalSeconds) {
+      details += `${Math.floor(settings.totalSeconds / 60)} min ${settings.totalSeconds % 60} sec `;
+    }
+    if (settings.workSeconds && settings.restSeconds) {
+      const workMin = Math.floor(settings.workSeconds / 60)
+      const workSec = settings.workSeconds % 60
+      const restMin = Math.floor(settings.restSeconds / 60)
+      const restSec = settings.restSeconds % 60
+
+      details += `(${workMin}min ${workSec} sec Work & ${restMin} sec ${restSec}sec Rest)`;
+    }
+
+    return details;
+  }
+
+// Total time calculation
+export const totalWorkoutTime = () => {
```

Comment:
hooks should follow the rules of hooks and use the naming convention: https://react.dev/learn/reusing-logic-with-custom-hooks#hook-names-always-start-with-use